@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -248,72 +235,89 @@ export class TextElementState {
-            for (const renderState of this.m_iconRenderStates as RenderState[]) {
-                renderState.updateFading(time, disableFading);
-            }
+            this.iconRenderState.updateFading(time, disableFading);
         }
     }
 
+    public get position(): THREE.Vector3 {
+        return this.element.position;
+    }
+
     /**
      * Initialize text and icon render states
      */
     private initializeRenderStates() {
         assert(this.m_textRenderState === undefined);
         assert(this.m_textLayoutState === undefined);
-        assert(this.m_iconRenderStates === undefined);
+        assert(this.m_iconRenderState === undefined);
 
         const { textFadeTime } = this.element;
         this.m_textRenderState = new RenderState(textFadeTime);
 
         const iconFadeTime = this.element.poiInfo?.technique.iconFadeTime;
-        if (this.element.type === TextElementType.LineMarker) {
-            this.m_iconRenderStates = new Array<RenderState>();
-            for (const _point of this.element.points as THREE.Vector3[]) {
-                const iconRenderStates = this.m_iconRenderStates as RenderState[];
-                const renderState = new RenderState(iconFadeTime);
-                iconRenderStates.push(renderState);
-            }
-            return;
-        }
 
-        if (this.element.type === TextElementType.PoiLabel) {
-            this.m_iconRenderStates = new RenderState(iconFadeTime);
+        if (
+            this.element.type === TextElementType.PoiLabel ||
+            this.element.type === TextElementType.LineMarker
+        ) {
+            this.m_iconRenderState = new RenderState(iconFadeTime);
         }
     }
 }
+
+export class LineMarkerElementState extends TextElementState {
+    private m_lineMarkerIndex: number;
+
+    constructor(element: TextElement, index: number) {
+        super(element);
+        this.m_lineMarkerIndex = index;
+    }
+
+    public get lineMarkerIndex(): number {
+        return this.m_lineMarkerIndex;
+    }
+
+    /**
+     * @override
+     */
+    public get position(): THREE.Vector3 {
+        return this.element.path
+            ? this.element.path[this.m_lineMarkerIndex]
+            : this.element.position;
+    }
+
+    /**
+     * @override
+     * Replaces given text element, inheriting its current state.
+     * The predecessor text element state is erased.
+     * @param predecessor - Text element state to be replaced.
+     */
+    replace(predecessor: TextElementState) {
+        super.replace(predecessor);
+        assert((predecessor as any).m_lineMarkerIndex !== undefined);
+        this.m_lineMarkerIndex = (predecessor as LineMarkerElementState).lineMarkerIndex!;
+    }
+}
+
+/**
+ * Test if a {@link TextElementState} is of type {@link LineMarkerElementState}
+ * @param state - Text element state to test.
+ */
+export function isLineMarkerElementState(state: TextElementState) {
+    return (state as any).m_lineMarkerIndex !== undefined;
+}
```

Comment:
To make it simpler, we can use TextElementState class for line markers as well and just pass and optional index to the constructor.
Then change the position getter to use the index if defined.